### PR TITLE
fix character and weapon select

### PR DESCRIPTION
**Pull request by srliao**
character and weapon select should now use pipeline data and stay up to date with core changes (as long as `go generate` is ran)

### CODE REVIEW

File: ui/packages/ui/src/Components/Select/weapons.tsx [hunk 1]
```diff
@@ -2,140 +2,9 @@ import { MenuItem } from "@blueprintjs/core";
...
-  "wastergreatsword",
-  "wavebreakersfin",
-  "whiteblind",
-  "whiteirongreatsword",
-  "whitetassel",
-  "windblumeode",
-  "wineandsong",
-  "wolfsgravestone",
-];
+export const weapons: IWeapon[] = valid_weapons.map(e => e)
```
Comment (k0l11) [inline ui/packages/ui/src/Components/Select/weapons.tsx:7]: why the map?

File: ui/packages/ui/src/Data/index.ts [hunk 2]
```diff
@@ -8,23 +9,60 @@ export interface CharDataMap {
...
+    case "WEAPON_CATALYST":
+      return "catalyst";
+    default:
+      return "unknown";
+  }
 }
 
-export function TravelerCheck(key: string) {
-  return key.startsWith("aether") || key.startsWith("lumine") || key.startsWith("traveler");
+export const valid_weapons: string[] = Object.keys(weaponPipelineData.data).map(
```
Comment (k0l11) [inline ui/packages/ui/src/Data/index.ts:50]: why the map?

File: ui/packages/ui/src/Components/Select/weapons.tsx (hunk 1 above)
Comment (srliao) [reply ui/packages/ui/src/Components/Select/weapons.tsx]: fixed

File: ui/packages/ui/src/Data/index.ts (hunk 2 above)
Comment (srliao) [reply ui/packages/ui/src/Data/index.ts]: fixed

**Review by k0l11 — APPROVED** [review]

**Pull request merged**